srt.go

```diff
@@ -34,17 +34,30 @@ func ReadFromSRT(i io.Reader) (o *Subtitles, err error) {
 	// Scan
 	var line string
 	var lineNum int
+
+	// For the Index Counter
+	saveIndex := true
+	var index int
+	var convErr error
+
 	var s = &Item{}
 	for scanner.Scan() {
 		// Fetch line
 		line = strings.TrimSpace(scanner.Text())
+
+		// Fetch Index
+		if saveIndex {
+			index, convErr = strconv.Atoi(strings.Replace(line, "\ufeff", "", -1))
+			if convErr == nil {
+				saveIndex = false
+			}
+		}
 		lineNum++
 
 		// Line contains time boundaries
 		if strings.Contains(line, srtTimeBoundariesSeparator) {
 			// Remove last item of previous subtitle since it's the index
 			s.Lines = s.Lines[:len(s.Lines)-1]
```

Comment:
I'd rather extract the index here with something along the lines of :

```go
// Extract index from previous item's last line
index := s.Lines[len(s.Lines) -1]
s.Lines = s.Lines[:len(s.Lines)-1]

...

// Init subtitle
s = &Item{}

// Parse index
s.Index, _ = strconv.Atoi(index)
```

Reply:
fixed with https://github.com/asticode/go-astisub/pull/22/commits/4adf45bbc0604cf1200febf067bd524d1cb1e451